### PR TITLE
Add /deletegroupchat command

### PR DESCRIPTION
**Pull request by Articuno-I**
/deletegroupchat function, which should be useful if you're finished using a room or want to change it from hidden to private, for example.

### CODE REVIEW

**Comment by zJoshz** [comment]
I'm pretty sure they don't want this.

**Comment by Zarel** [comment]
It's actually nice for admins, but discussions say that ROs shouldn't be able to do it.

**Comment by panpawn** [comment]
If it's going to be for admins, why not just make a clause for /deregisterchatroom that checks if it's a personal room and then if so, go from there?

**Comment by Zarel** [comment]
Honestly, just `/deleteroom`

File: commands.js [hunk 1]
```diff
@@ -375,6 +375,19 @@ var commands = exports.commands = {
...
+	deletegroupchat: function (target, room, user) {
+		var id = toId(target);
+		if (!id) id = toId(room);
+		var targetRoom = Rooms.search(id);
+		target = targetRoom.title || targetRoom.id;
+		if (!targetRoom) return this.sendReply("The room '" + target + "' doesn't exist.");
+		if (!targetRoom.isPersonal) return this.sendReply("The room '" + target + "' is not a group chat.");
+		if (targetRoom.auth[user.userid] !== '#' && !this.can('makeroom')) return;
+		targetRoom.destroy();
+		return this.sendReply("The room '" + target + "' has been deleted.");
```
Comment (panpawn) [inline commands.js]: If the room is being destroyed and a room owner used it in the room, how would they get this return message?

(Or is this slower than the line above it...?)

**Comment by Articuno-I** [comment]
I was thinking it'd be useful for ROs to have if they want to remake the room with different public/hidden/private settings, it's also somewhat useful if they're just not using the room any more due to taking a little bit of stress off the server for 40 minutes.

@panpawn I forgot it'd send that reply if you used the command in the room you're going to delete, I'll change that to only work if it's a different room now.

**Comment by Zarel** [comment]
The 40 minutes server stress is nothing, and other people might want to keep talking. Users in general suck at realizing how much the things they do hurt other people.

Changing public/private settings: the command `/privateroom` already does this without needing to restart.

**Comment by Articuno-I** [comment]
That's a fair point. In that case, should /privateroom be changed to allow owners of groupchats to use the command?

**Comment by Zarel** [comment]
In theory, it already does. Maybe there's a bug involved?

**Comment by AustinXII** [comment]
@Zarel I think he means to turn it off and on without having to make a new room.

**Comment by zJoshz** [comment]
Yeah, as of now there is no way to change it from private/public/hidden besides deleting and recreating the room altogether.

**Comment by Zarel** [comment]
Fixed.

**Comment by Articuno-I** [comment]
So there's been some changes to how groupchats work, do you still want the command changed to admins only and /deleteroom?

**Comment by AustinXII** [comment]
I agree with @panpawn  about refactoring /deregisterchatroom I don't see a point in adding two commands that basically do the same thing.

**Comment by Zarel** [comment]
Done in bda75f3ab0

**Pull request closed without merging**